src/langauge-servers/configStore.ts

```diff
@@ -1,95 +1,319 @@
+import {findGoLocation, getGoVersion} from "./goUtils";
+import { findJavaHomes, getJavaVersion } from "./javaUtils";
+import {findPythonLocation, getPythonVersion, PIP_FILENAME} from "./pythonUtils";
+
+const isWindows =  !_.isEmpty(process.platform.match("win32"));
+
+export const logger = getLogger("langServer");
+const langServerExecFolder = path.join(getLibraryFolder(), "languageServers");
+
+const JavaLangServerDownloadURL = "https://get.rookout.com/Language-Servers/Rookout-Java-Language-Server.jar";
+export const javaLangServerJarLocation = path.join(langServerExecFolder, "Rookout-Java-Language-Server.jar");
+export const langServersNpmInstallationLocation = path.join(langServerExecFolder, "npm_modules");
+export const javascriptLangServerExecLocation = path.join(langServersNpmInstallationLocation, "node_modules", "quick-lint-js", "quick-lint-js.exe");
+export const typescriptLangServerExecLocation = path.join(
+    langServersNpmInstallationLocation, "node_modules", "typescript-language-server", "lib", "cli.js");
+
+export const supportedLanguageServers: string[] = ["java", "python", "go", "typescript"];
+
+export const minimumLanguageVersions: {[language: string]: string} = {
+    java: "13",
+    python: "3.7",
+    go: "1.17"
+};
+const languageToGetVersionFunction: {[language: string]: (location: string) => string} = {
+    java: getJavaVersion,
+    python: getPythonVersion,
+    go: getGoVersion
+};
+
+export const maximumLanguageVersions: {[language: string]: string} = {
+};
+
+const LANGUAGE_STORE_ENABLE_KEYS: {[language: string]: string} = {
+    java: "enable-java-server",
+    python: "enable-python-server",
+    go: "enable-go-server",
+    typescript: "enable-typescript-server"
+};
+
+const LANGUAGE_STORE_LOCATION_KEYS: {[language: string]: string} = {
+    java: "java-home-location",
+    python: "python-location",
+    go: "go-location",
+};
 
 class LangServerConfigStore {
-    private store: IStore
-    public isDownloadingJavaJar: boolean = false
-    public jdkLocation: string
+    private static installPythonLanguageServer(pythonLocation: string) {
+        cp.execFileSync(PIP_FILENAME, ["install", "python-lsp-server[all]"], { cwd: pythonLocation, encoding: "utf-8" });
+    }
+
+    public serverLocations: {[language: string]: string} = {
+        java: "",
+        python: "",
+        go: ""
+    };
+    public enabledServers: {[language: string]: boolean} = {
+        java: false,
+        python: false,
+        go: false,
+        typescript: false,
+        javascript: false
+    };
+
+    private store: IStore;
 
     constructor() {
-        this.store = getStoreSafe()
-        this.ensureLangServerExecFolderExists()
+        this.store = getStoreSafe();
+        this.ensureLangServerExecFolderExists();
 
-        if (!this.doesJavaJarExist()) {
-            this.downloadJavaLangServer()
-        }
+        this.serverLocations["java"] = this.store.get(LANGUAGE_STORE_LOCATION_KEYS["java"], "");
```

Comment:
Here you can loop over the values of the languages enum you've made instead of hardcoding every single language

```javascript
const serverLocations = Object.values(LanguagesEnum).forEach(language => {
     this.serverLocations[language] = this.store.get(languageLocationKey(language), '');
     this.enabledServers[language] = this.store.get(languageEnabledKey(language), 'false') === 'true';
})
```

Where `languageLocationKey` and `languageEnabledKey` could be functions that get you the appropriate storage key with the pattern above